backend/src/activitypub/objects/index.ts

```diff
@@ -179,3 +181,71 @@ WHERE objects.${key}=?
 		originalObjectId: result.original_object_id,
 	} as Object
 }
+
+/** Is the given `value` an ActivityPub Object? */
+export function isObject(value: unknown): value is Object {
+	return value !== null && typeof value === 'object'
+}
+
+/** Sanitizes the ActivityPub Object `properties` prior to being stored in the DB. */
+export async function sanitizeObjectProperties(properties: unknown): Promise<Object> {
+	if (!isObject(properties)) {
+		throw new Error('Invalid object properties. Expected an object but got ' + JSON.stringify(properties))
+	}
+	const sanitized: Object = {
+		...properties,
+	}
+	if ('content' in properties) {
+		sanitized.content = await sanitizeContent(properties.content as string)
+	}
+	if ('name' in properties) {
+		sanitized.name = await sanitizeName(properties.name as string)
+	}
+	return sanitized
+}
+
+/**
+ * Sanitizes the given string as ActivityPub Object content.
+ *
+ * This sanitization follows that of Mastodon
+ *  - convert all elements to `<p>` unless they are recognized as one of `<p>`, `<span>`, `<br>` or `<a>`.
+ *  - remove all CSS classes that are not micro-formats or semantic.
+ *
+ * See https://docs.joinmastodon.org/spec/activitypub/#sanitization
+ */
+export async function sanitizeContent(unsafeContent: string): Promise<string> {
+	return await contentRewriter.transform(new Response(unsafeContent)).text()
+}
+
+/**
+ * Sanitizes given string as an ActivityPub Object name.
+ *
+ * This sanitization removes all HTML elements from the string leaving only the text content.
+ */
+export async function sanitizeName(dirty: string): Promise<string> {
+	return await nameRewriter.transform(new Response(dirty)).text()
```

Comment:
```suggestion
export async function sanitizeName(unsafeName: string): Promise<string> {
	return await nameRewriter.transform(new Response(unsafeName)).text()
```
(for consistency with the `sanitizeContent` function)